### ISSUE TITLE
Route reflector Docker image does not support peering with non-Calico neighbors

### ISSUE THREAD

**Issue by rfc1459** [projectcalico/bird#18]
From a cursory glance at [bird.cfg.template](https://github.com/projectcalico/calico-bird/blob/6a7902321e29363eb24936ed9dfcbed4d88f7ac0/build_routereflector/node_filesystem/templates/bird.cfg.template), it seems that the dockerized route reflector is capable only of peering with the following kind of neighbors:
- other RRs in a full mesh
- other Calico nodes as a global peer
- other Calico nodes as a node-specific peer

This means that (as it is) the RR image is not deployable in a scenario with a border gateway because it won't be able to peer with it.

**Comment by rikatz** [comment]
@rfc1459 Have you tried to add it manually to bird.cfg?

I'm facing a similar problem here, and added manually it. But I've realized that when adding some new Nodes to Calico Cluster, the bird.cfg file from my route reflector does not get updated. 

Maybe after correcting this 'bug' we could create a bgpPeer as the following, and get the RR updated:

```yaml
apiVersion: v1
kind: bgpPeer
metadata:
  peerIP: 10.15.28.161 
  scope: node
  node: router
spec:
  asNumber: 64512
```

And least, if this does not work you can manually create a RR as described in docs, and set your very own confd to update this:

http://docs.projectcalico.org/v2.1/usage/bird-rr-config

**Comment by rikatz** [comment]
By the way, I'm facing this same problem trying to configure a FreeBSD + Carp as my entrance router.

**Comment by rikatz** [comment]
OK, I could figure out that when you create a 'bgpPeer' resource, it creates an entry in ETCD as /calico/bgp/v1/host/router/peer_v4 that breaks the template used to RouteReflector.

Then I've figured out that to enable what we want, just need to create a 'Node' Resource, as the following:

```yaml
apiVersion: v1
kind: node
metadata:
  name: router
spec:
  bgp:
    asNumber: 64512
    ipv4Address: 10.15.28.161
```

After that, it was possible to pair my OpenBGP router with calico:

```
bgpctl show
Neighbor                   AS    MsgRcvd    MsgSent  OutQ Up/Down  State/PrfRcvd
AS64512-rr              64512          6          3     0 00:00:25      3
```

```
 bgpctl show ip bgp
flags: * = Valid, > = Selected, I = via IBGP, A = Announced, S = Stale
origin: i = IGP, e = EGP, ? = Incomplete

flags destination          gateway          lpref   med aspath origin
I*>   192.168.140.64/26    10.15.28.152       100     0 i
I*>   192.168.186.192/26   10.15.28.153       100     0 i
I*>   192.168.196.128/26   10.15.28.151       100     0 i
```

**Comment by robbrockbank** [comment]
Hi @rikatz - glad you managed to find a way to work around things.

The RR image is somewhat experimental in that we primarily use it for our ST framework so we didn't build it out with real deployments in mind.  That said, we'd happily take PRs to improve it, and longer term it is on my radar to improve our general BGP configuration.

Just as an FYI, I've moved the doc to our main docs now and fixed up the calicoctl commands since the syntax has changed:

http://docs.projectcalico.org/master/usage/routereflector/calico-routereflector

**Comment by rikatz** [comment]
@robbrockbank Thanks. I'll take a look at that when I come back from my vacations, and will make a PR to improve that error handling in the CONFD from RR

**Comment by caseydavenport** [comment]
This has been fixed in Calico v3.3 by adding RR capabilities to the calico/node container. 

calico/routereflector is being deprecated, so I'm going to close this.
